tests/extension/oneapi-current-device/current_device.cpp

```diff
@@ -0,0 +1,136 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2023 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#include "../../common/common.h"
+#include <condition_variable>
+#include <mutex>
+#include <thread>
+#include <vector>
+
+namespace current_device {
+#ifdef SYCL_EXT_ONEAPI_CURRENT_DEVICE
+
+sycl::device get_device_to_set(const bool is_need_to_reallocate) {
+  static std::vector<sycl::device> devices;  // = sycl::device::get_devices();
+  if (is_need_to_reallocate) {
+    devices = sycl::device::get_devices();
+  }
+  for (const auto device : devices) {
+    // remove device from the list of devices
+    devices.erase(std::remove(devices.begin(), devices.end(), device),
+                  devices.end());
+    return device;
+  }
+  SKIP("Test did not find any device to set as current");
+}
+
+#endif
+
+TEST_CASE(
+    "Test for \"get_current_device\" extension, get default device, call "
+    "function and compare devices ") {
+#ifndef SYCL_EXT_ONEAPI_CURRENT_DEVICE
+  SKIP("SYCL_EXT_ONEAPI_CURRENT_DEVICE is not defined");
+#else
+  if (sycl::device::get_devices().size() < 1) {
+    SKIP("Test requires at least one device");
+  }
+  const sycl::device default_device{sycl::default_selector()};
+  const auto current_device =
+      sycl::ext::oneapi::experimental::this_thread::get_current_device();
+  CHECK(
+      std::is_same_v<std::remove_cv_t<decltype(current_device)>, sycl::device>);
+  CHECK(default_device == current_device);
+#endif
+}
+
+TEST_CASE(
+    "Test for \"set_current_device and get_current_device\" extension, check "
+    "that get function returns the device set by set function") {
+#ifndef SYCL_EXT_ONEAPI_CURRENT_DEVICE
+  SKIP("SYCL_EXT_ONEAPI_CURRENT_DEVICE is not defined");
+#else
+  const auto device_to_set = current_device::get_device_to_set(true);
+  sycl::ext::oneapi::experimental::this_thread::set_current_device(
+      device_to_set);
+  const auto current_device =
+      sycl::ext::oneapi::experimental::this_thread::get_current_device();
+  CHECK(
+      std::is_same_v<std::remove_cv_t<decltype(current_device)>, sycl::device>);
+  CHECK(device_to_set == current_device);
```

Comment:
We should also check all other available devices in the loop. The test currently checks only first available device from `sycl::device::get_devices()`